lib/logstash/inputs/udp.rb

```diff
@@ -39,15 +39,24 @@ def initialize(params)
 
   public
   def register
-    @udp = nil
+    @logger.info("Starting UDP listener", :address => "#{@host}:#{@port}")
+
+    if @udp && ! @udp.closed?
+      @udp.close
+    end
+
+    @udp = UDPSocket.new(Socket::AF_INET)
+    @udp.bind(@host, @port)
+
+    @input_to_worker = SizedQueue.new(@queue_size)
```

Comment:
This change a bit the behavior of the plugin.
If there is an exception done at the socket level we will never recreate the server.